tests/test_manifest.py

```diff
@@ -837,19 +837,18 @@ def test_parse_multiple_manifest_files(manifest_repo):
 
     # Manifest.from_file() should be also usable with another_yml.
     # The project hierarchy in its return value should still be rooted
-    # in the topdir, but the resulting ManifestProject does not have a
-    # path, because it's not set in the file, and we're explicitly not
-    # comparing its path to manifest.path.
+    # in the topdir, but the resulting ManifestProject will have a path
+    # identical to the path of the manifest file.
     #
     # However, the project hierarchy *must* be rooted at topdir.
     manifest = Manifest.from_file(source_file=another_yml)
     assert len(manifest.projects) == 3
     assert manifest.topdir is not None
     assert PurePath(manifest.topdir) == PurePath(topdir)
     mproj = manifest.projects[0]
-    assert mproj.path is None
-    assert mproj.abspath is None
-    assert mproj.posixpath is None
+    assert mproj.path == topdir
```

Comment:
This doesn't look right. How could the manifest project have the same path as the top level directory of the workspace?

By definition, it is always in a directory of topdir.

Reply:
If you look at the code:
https://github.com/zephyrproject-rtos/west/blob/95cd8ff2782c694c78f29391890216b2fef9f0db/tests/test_manifest.py#L804
https://github.com/zephyrproject-rtos/west/blob/95cd8ff2782c694c78f29391890216b2fef9f0db/tests/test_manifest.py#L844
```
    another_yml = topdir / 'another.yml'
...
    manifest = Manifest.from_file(source_file=another_yml)
....
```
so as you see, this test case uses a specific yaml file located directly in the topdir, and NOT a manifest file located in `manifest.path`.
Because this is provided directly as `Manifest.from_file(source_file=another_yml)` then the manifest path to `another_yml` is identical to `topdir` in this particular case.

Reply:
The results are nonsense in that case. The manifest.path attribute should be `None` because there is no manifest repository. There can't be a path to a manifest repository if there is no such repository.

Taking another look at the comments in this test, it even documents this intent:

```
    # [...] the resulting ManifestProject does not have a
    # path, because it's not set in the file, and we're explicitly not
    # comparing its path to manifest.path.
```

Reply:
> . The manifest.path attribute should be None because there is no manifest repository.

Well, the manifest path is a path. You don't need the manifest project/folder to be a repository, but it is convenient.

But as soon as `Manifest.from_file()` and `Manifest.from_file(source_file=<default_west_yml>)` produces different results on the same input, then I think consistency breaks.
See: https://github.com/zephyrproject-rtos/west/pull/422#issuecomment-675321128